src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -460,11 +460,14 @@ const DataProcessingPage = ({ experimentId, experimentData }) => {
                       <Button
                         disabled={prefiltered}
                         data-testid='enableFilterButton'
-                        onClick={() => {
-                          dispatch(setQCStepEnabled(
-                            currentStep.key, !stepEnabled,
-                          ));
-                          dispatch(saveProcessingSettings(experimentId, currentStep.key));
+                        onClick={async () => {
+                          dispatch(saveProcessingSettings(experimentId, currentStep.key)).then(() => {
+                            if (processingConfig.meta.saveSettingsError === false) {
```

Comment:
in this if instead of checking `processingConfig.meta.saveSettingsError` can you not do that check somewhere from what `then` returns - `blabla.then((res)=>if(!res.ok))` or something like that?

Reply:
I reviewed this with Iva and it turns out returning things in actions is not the correct way to go. I added back the `if` way because it's the recommended way of handling things in react/redux https://stackoverflow.com/a/58299220 https://exportknowledge.com/article/error-handling-with-react